microsoft/testsuites/vm_extensions/azure_keyvault_vm_extension.py

```diff
@@ -0,0 +1,231 @@
+        )
+
+        log.info(f"Created Key Vault {keyvault_result.properties.vault_uri}")
+
+        certificates_secret_id: List[str] = []
+        for cert_name in ["Cert1", "Cert2"]:
+            certificate_secret_id = create_certificate(
+                platform=platform,
+                vault_url=keyvault_result.properties.vault_uri,
+                log=log,
+                cert_name=cert_name,
+            )
+            log.info(f"Certificates created. Cert ID: {certificate_secret_id}, ")
+            assert_that(certificate_secret_id).described_as(
+                "First certificate created successfully"
+            ).is_not_none()
+            certificates_secret_id.append(certificate_secret_id)
+
+        # Extension
+        extension_name = "KeyVaultForLinux"
+        extension_publisher = "Microsoft.Azure.KeyVault"
+        extension_version = "2.0"
+        settings = {
+            "secretsManagementSettings": {
+                "autoUpgradeMinorVersion": True,
+                "enableAutomaticUpgrade": True,
+                "pollingIntervalInS": "360",
+                "certificateStoreLocation": "/var/lib/waagent/Microsoft.Azure.KeyVault",
+                "observedCertificates": [
+                    certificates_secret_id[0],
+                    certificates_secret_id[1],
+                ],
+            }
+        }
+        extension = node.features[AzureExtension]
+        extension_result = extension.create_or_update(
+            name=extension_name,
+            publisher=extension_publisher,
+            type_=extension_name,
+            type_handler_version=extension_version,
+            auto_upgrade_minor_version=True,
+            enable_automatic_upgrade=True,
+            settings=settings,
+        )
+        assert_that(extension_result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        # Rotate certificates
+        rotate_certificates(
+            log,
+            vault_url=keyvault_result.properties.vault_uri,
+            platform=platform,
+            cert_name_to_rotate="Cert1",
+        )
+
+        _check_system_status(node, log)
+
+        # Deleting the certificates after the test
+        for cert_name in ["Cert2", "Cert1"]:
+            delete_certificate(
+                platform=platform,
+                vault_url=keyvault_result.properties.vault_uri,
+                cert_name=cert_name,
+                log=log,
+            )
+
+            retries = 2
+            for _ in range(retries):
+                certificate_exists = check_certificate_existence(
+                    vault_url=keyvault_result.properties.vault_uri,
+                    cert_name=cert_name,
+                    log=log,
+                    platform=platform,
+                )
+                if not certificate_exists:
+                    break
+                time.sleep(5)  # wait for 5 seconds before retrying
```

Comment:
5 seconds is too long to check. Try 10 times every 1 second. It still allow 10 seconds before timeout, but won't waste time more than 1 second.